src/mongocxx/write_concern.hpp

```diff
@@ -183,7 +204,14 @@ class MONGOCXX_API write_concern {
     ///
     /// @return The number of required nodes.
     ///
-    std::int32_t nodes() const;
+    stdx::optional<std::int32_t> nodes() const;
```

Comment:
This is optional because the special kDefault, kMajority, and kUnacknowledged live in the same C int32 as the nodes, so only one of `nodes()` or `acknowledge_level()` will ever be set.